src/SOIL2.m

```diff
@@ -1,7 +1,64 @@
-function [hh, COR, CORh, Theta_V, Theta_g, Se, KL_h, Theta_LL, DTheta_LLh, KfL_h, KfL_T, hh_frez, Theta_UU, DTheta_UUh, Theta_II] = SOIL2(SoilConstants, SoilVariables, hh, COR, hThmrl, NN, NL, TT, Tr, Hystrs, XWRE, Theta_s, IH, KIT, Theta_r, Alpha, n, m, Ks, Theta_L, h, Thmrlefc, POR, Theta_II, CORh, hh_frez, h_frez, SWCC, Theta_U, XCAP, Phi_s, RHOI, RHOL, Lamda, Imped, L_f, g, T0, TT_CRIT, KfL_h, KfL_T, KL_h, Theta_UU, Theta_LL, DTheta_LLh, DTheta_UUh, Se)
+function SoilVariables = SOIL2(KIT, L_f, SoilVariables, VanGenuchten)
 
-    if hThmrl == 1
+    % these lines can be removed after issue 181
+    % get model settings
+    ModelSettings = io.getModelSettings();
+
+    NN = ModelSettings.NN;
+    NL = ModelSettings.NL;
+    SWCC = ModelSettings.SWCC;
+    Thmrlefc = ModelSettings.Thmrlefc;
+    T0 = ModelSettings.T0;
+    Tr = ModelSettings.Tr;
+    hThmrl = ModelSettings.hThmrl;
+    Hystrs = ModelSettings.Hystrs;
+
+    Theta_s = VanGenuchten.Theta_s;
+    Theta_r = VanGenuchten.Theta_r;
+    Alpha = VanGenuchten.Alpha;
+    n = VanGenuchten.n;
+    m = VanGenuchten.m;
+
+    POR = SoilVariables.POR;

```

Comment:
Instead of unpacking the struct, and then updating it. How about copying it with an alias (e.g. `SV` or `SVar`)? This would simplify this file a bit. I do agree that
```matlab
SoilVariables.Theta_V(i, ND) = SoilVariables.POR(i) - SoilVariables.Theta_UU(i, ND);
```
is a bit much. But this seems OK to me:
```matlab
SV.Theta_V(i, ND) = SV.POR(i) - SV.Theta_UU(i, ND);
```

Reply:
agreed. But the problem is that some of the variables are still used as global in other scripts. Then, I need to unpack the structure. When all globals are removed. this refactoring can be applied.

Reply:
But this happens in a function, and it has no global declarations, so that shouldn't be required.